pkg/opensearch/client/search_request_test.go

```diff
@@ -6,457 +6,363 @@ import (
-						So(childAgg.Key, ShouldEqual, "2")
-						So(childAgg.Aggregation.Type, ShouldEqual, "terms")
-
-						childChildAgg := childAgg.Aggregation.Aggs[0]
-						So(childChildAgg.Key, ShouldEqual, "3")
-						So(childChildAgg.Aggregation.Type, ShouldEqual, "date_histogram")
-					})
+		t.Run("and adding multiple top level aggs, When building search request", func(t *testing.T) {
+			aggBuilder := b.Agg()
+			aggBuilder.Terms("1", "@hostname", nil)
+			aggBuilder.DateHistogram("2", "@timestamp", nil)
+
+			sr, err := b.Build()
+			assert.NoError(t, err)
+
+			t.Run("Should have 2 top level aggs", func(t *testing.T) {
+				aggs := sr.Aggs
+				assert.Len(t, aggs, 2)
+				assert.Equal(t, "1", aggs[0].Key)
+				assert.Equal(t, "terms", aggs[0].Aggregation.Type)
+				assert.Equal(t, "2", aggs[1].Key)
+				assert.Equal(t, "date_histogram", aggs[1].Aggregation.Type)
+			})
 
-					Convey("When marshal to JSON should generate correct json", func() {
-						body, err := json.Marshal(sr)
-						So(err, ShouldBeNil)
-						json, err := simplejson.NewJson(body)
-						So(err, ShouldBeNil)
+			t.Run("When marshal to JSON should generate correct json", func(t *testing.T) {
+				body, err := json.Marshal(sr)
+				assert.NoError(t, err)
+				json, err := simplejson.NewJson(body)
+				assert.NoError(t, err)
 
-						topAgg := json.GetPath("aggs", "1")
-						So(topAgg.GetPath("terms", "field").MustString(), ShouldEqual, "@hostname")
+				assert.Len(t, json.Get("aggs").MustMap(), 2)
+				assert.Equal(t, "@hostname", json.GetPath("aggs", "1", "terms", "field").MustString())
+				assert.Equal(t, "@timestamp", json.GetPath("aggs", "2", "date_histogram", "field").MustString())
+			})
+		})
 
-						childAgg := topAgg.GetPath("aggs", "2")
-						So(childAgg.GetPath("terms", "field").MustString(), ShouldEqual, "@app")
+		t.Run("and adding two top level aggs with child agg", func(t *testing.T) {
+			aggBuilder := b.Agg()
+			aggBuilder.Histogram("1", "@hostname", func(a *HistogramAgg, ib AggBuilder) {
+				ib.DateHistogram("2", "@timestamp", nil)
+			})
+			aggBuilder.Filters("3", func(a *FiltersAggregation, ib AggBuilder) {
+				ib.Terms("4", "@test", nil)
+			})
+		})
 
-						childChildAgg := childAgg.GetPath("aggs", "3")
-						So(childChildAgg.GetPath("date_histogram", "field").MustString(), ShouldEqual, "@timestamp")
-					})
+		t.Run("and adding top level agg with child agg with child agg", func(t *testing.T) {
+			aggBuilder := b.Agg()
+			aggBuilder.Terms("1", "@hostname", func(a *TermsAggregation, ib AggBuilder) {
+				ib.Terms("2", "@app", func(a *TermsAggregation, ib AggBuilder) {
+					ib.DateHistogram("3", "@timestamp", nil)
 				})
 			})
+		})
 
-			Convey("and adding bucket and metric aggs", func() {
-				aggBuilder := b.Agg()
-				aggBuilder.Terms("1", "@hostname", func(a *TermsAggregation, ib AggBuilder) {
-					ib.Terms("2", "@app", func(a *TermsAggregation, ib AggBuilder) {
+		t.Run("and adding bucket and metric aggs", func(t *testing.T) {
```

Comment:
I might be misreading the diff, but I think the nested portions of this test are missing